File: entregable 4/areEqual.py
def main():
    palabra1 = input()
    palabra2 = input()
    if palabra1 == palabra2 or areEqual(palabra1, palabra2):
        print ("YES")
    else:
        print ("NO")

def areEqual(a, b):
    if len(a) == 1 and len(b) == 1:
        if a == b:
            return True
        else:
            return False
    else:
        if len(a) % 2 == 0:
            mitad = len(a) // 2
            a1, a2 = a[:mitad], a[mitad:]
            b1, b2 = b[:mitad], b[mitad:]
            return (areEqual(a1, b2) and areEqual(a2, b1) or areEqual(a1, b1) and areEqual(a2, b2))
        else:
            if(a == b):
                return True
            else:
                return False
        
if __name__ == "_main_":
    main()

File: entregable 4/areEqual.cpp
#include <iostream>
#include <string>

bool areEqual(std::string a, std::string b) {
    if (a.length() == 1 && b.length() == 1) {
        return a == b;
    } else {
        if (a.length() % 2 == 0) {
            int mid = a.length() / 2;
            std::string a1 = a.substr(0, mid);
            std::string a2 = a.substr(mid);
            std::string b1 = b.substr(0, mid);
            std::string b2 = b.substr(mid);
            return (areEqual(a1, b2) && areEqual(a2, b1)) || (areEqual(a1, b1) && areEqual(a2, b2));
        } else {
            return a == b;
        }
    }
}

int main() {
    std::string palabra1, palabra2;
    std::cin >> palabra1 >> palabra2;

    if (palabra1 == palabra2 || areEqual(palabra1, palabra2)) {
        std::cout << "YES" << std::endl;
    } else {
        std::cout << "NO" << std::endl;
    }

    return 0;
}
